src/Imaginocean.jl

```diff
@@ -210,23 +208,25 @@ function heatsphere!(axis::Axis3, field::Field, k_index=1; kwargs...)
     return axis
 end
 
-function heatmap!(ax::Axis, field::Field, k=1; kwargs...)
+"""
+    function Makie.convert_arguments(P::SurfaceLike, field::Field, depth_level::Int64)
+Convert an Oceananigans.jl `Field` at `depth_level` to arguments that can be plotted as a
+`SurfaceLike` type in Makie.jl.
+"""
+function Makie.convert_arguments(P::SurfaceLike, field::Field, depth_level::Int64)
+
     LX, LY, LZ = location(field)
     grid = field.grid
 
     _, (λvertices, φvertices) = get_lat_lon_nodes_and_vertices(grid, LX(), LY(), LZ())
 
-    quad_points = vcat([Point2.(λvertices[:, i, j], φvertices[:, i, j]) for i in axes(λvertices, 2), j in axes(λvertices, 3)]...)
-    quad_faces = vcat([begin; j = (i-1) * 4 + 1; [j j+1  j+2; j+2 j+3 j]; end for i in 1:length(quad_points)÷4]...)
-
-    colors_per_point = vcat(fill.(vec(interior(field, :, :, k)), 4)...)
-
-    mesh!(ax, quad_points, quad_faces; color = colors_per_point, shading = false, kwargs...)
+    quad_points = vcat([Point2f.(λvertices[:, i, j], φvertices[:, i, j]) for i in axes(λvertices, 2), j in axes(λvertices, 3)]...)
 
-    xlims!(ax, (-180, 180))
-    ylims!(ax, (-90, 90))
+    lons = unique([quad_points[i][1] for i ∈ eachindex(quad_points)])
+    lats = unique([quad_points[i][2] for i ∈ eachindex(quad_points)])
```

Comment:
I plotted it and looked OK.

But what do these lines do?

Reply:
These lines collect the cartesian grid points, converted from the longitude and latitude grid, for the data in the `Field`. Not sure if it is the best way and am just having a look at it now. Maybe can just call `lat_lon_to_cartesian` to get these cartesian grid points?